athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/storage/datasource/StorageDatasourceFactory.java

```diff
@@ -0,0 +1,50 @@
+/*-
+ * #%L
+ * athena-storage-api
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs.storage.datasource;
+
+import com.amazonaws.athena.connectors.gcs.storage.StorageMetadata;
+import com.amazonaws.athena.connectors.gcs.storage.StorageMetadataImpl;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class StorageDatasourceFactory
```

Comment:
This entire file can now be deleted, since its sole purpose was to just construct a `StorageMetadataImpl` instance. The code that calls this can just directly new the `StorageMetadata` class now.

Reply:
Removed.